Standards/scs-0101-v1-entropy.md

```diff
@@ -5,114 +5,127 @@ status: Draft
+file `rng_available` lists availabe HRNGs while the file `rng_current`
+contains the HRNG currently used.
 
-##### 1.2.3 Changes since Linux 5.17, 5.18
+In summary, with current kernels and CPUs entropy in virtual instances
+is readily available to a sufficient degree. In addition, the host's
+entropy sources can be injected using `virtio-rng` if so desired, e.g.,
+to enable access to a HRNG.
 
-Jason A. Donenfeld has rewritten the RNG Number generator for Linux, which is
-replacing the very old sha-1 with blake2 algorithm, "random use computational
-hash for entropy extraction" the full explanation can be found here:
-[linux-rng-5.17-18](https://web.archive.org/web/20230321040526/https://www.zx2c4.com/projects/linux-rng-5.17-5.18/).
+## Motivation
 
-This RNG improvements make some workarounds obsolete. For example haveged
-should not be use anymore. Rng-tools can continue to be used. Rng-tools bridge
-the hardware number generators that support RDRAND and RDSEED as they support
-HWRNG in modern Intel and AMD processors.
+As stated above, good sources of entropy are paramount for many
+important applications. This standard ensures that sufficient entropy
+will be available in virtual instances.
 
-These patches have now also arrived in the upstream LTS Kernels.
+## Entropy in SCS clouds
 
-The behavior will be as follows:
+### Flavors
+
+It is recommended that all flavors have the following attribute:
 
 ```console
- $cat /proc/sys/kernel/random/entropy_avail
-  256
+hw_rng:allowed=True
 ```
 
-Because the blake2 is used algorithm the entropy count is sufficient.
-The rng-tools brings up the software rngtest. With rngtest the entropy
-can be checked.
+The following attributes are optional:
 
 ```console
- $cat /dev/random | rngtest -c 1000
+hw_rng:rate_bytes - The allowed amount of bytes for the the guest
+    to read from the host's entropy per period.
+hw_rng:rate_period - Sets the duration of a read period in seconds.
 ```
 
-### 1.3 Entropy in SCS Clouds
-
-#### 1.3.1 Flavors
+### Images
 
-All flavors need to have the relevant attributes activated:
+It is recommended to use images having a kernel (patch level) version 5.18
+or up. This condition is already satisfied by every mandatory image defined
+in the [Image Metadata Standard](https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0102-v1-image-metadata.md).
 
-```console
-hw_rng:allowed=True
-```
+It is recommended that images activate the attribute `hw_rng_model: virtio`.
 
-optional:
+The daemon `rngd` must be installed (usually from `rng-tools`
+or `rng-utils`).
 
-```console
-hw_rng:rate_bytes - The allowed amount of bytes for the the guest
-to read from the host’s entropy per period.
+The user may choose to use the `virtio-rng` device via `rngd`.
 
-hw_rng:rate_period
-```
+### Compute nodes
 
-#### 1.3.2 Images
+Compute nodes must use CPUs that offer RDRAND/RDSEED, and these
+instructions may not be filtered by the hypervisor. If this requirement
```

Comment:
this will prevent CSPs from deploying ARM64 based compute nodes once we offer this functionality (which we will). Imho we should write this in a way that does not require us to revise it when we do that.